pysnooper/pysnooper.py

```diff
@@ -64,15 +65,36 @@ def snoop(output=None, variables=(), depth=1, prefix='', overwrite=False):
     if truncate is None and overwrite:
         raise Exception("`overwrite=True` can only be used when writing "
                         "content to file.")
-    def decorate(function):
+
+    def decorator(function):
         target_code_object = function.__code__
         tracer = Tracer(target_code_object=target_code_object, write=write,
                         truncate=truncate, variables=variables, depth=depth,
                         prefix=prefix, overwrite=overwrite)
 
-        def inner(function_, *args, **kwargs):
+        async def coroutine_wrapper(_, *args, **kwargs):
+            with tracer:
+                return await function(*args, **kwargs)
+
+        def generator_wrapper(_, *args, **kwargs):
+            it = iter(function(*args, **kwargs))
+            try:
+                while True:
+                    with tracer:
+                        yield next(it)
+            except StopIteration as e:
```

Comment:
I think it would be better if the only thing this try/except was wrapping was `next(it)`.

Reply:
Nice catch